cmake/Modules/GeneratePkgConfig/generate-pkg-config.cmake.in

```diff
@@ -1,52 +1,146 @@
 	string(REPLACE ";" " " _tmp_list "${_tmp_list}")
-	set(${_result} "${_tmp_list}" PARENT_SCOPE)
+
+	set(${ARG_RESULT_VAR} "${_tmp_list}" PARENT_SCOPE)
 endfunction()
 
 # Helper function for splitting full library paths into [dir, name] and merging repetitive dir entries
-function(split_library_dirs _libraries _base_library_dir _library_dirs_var _library_names_var)
-	set(libdirs "${_base_library_dir}")
-	set(libs "")
-	foreach (l IN LISTS _libraries)
-		get_filename_component(lDir "${l}" DIRECTORY)
+# Arguments:
+# - RESULT_LIB_DIRS: variable that will contain the resulting list of library directories in the parent scope
+# - RESULT_LIB_NAMES: variable that will contain the resulting list of library names in the parent scope
+# - LIBS: A CMake list of interface link libraries of the target
+# - BASE_LIB_DIR: base library directory
+function(split_library_dirs)
+
+	set(options "")
+	set(oneValueArgs RESULT_LIB_DIRS RESULT_LIB_NAMES LIBS BASE_LIB_DIR)
+	set(multiValueArgs "")
+
+	cmake_parse_arguments(
+		PARSE_ARGV 0
+		ARG
+		"${options}" "${oneValueArgs}" "${multiValueArgs}"
+	)
+
+	set(_libdirs "${ARG_BASE_LIB_DIR}")
+	set(_libs_result "")
+	foreach (lib IN LISTS ARG_LIBS)
+		# TODO: replace with cmake_path(...) in CMake >= 3.19
+		get_filename_component(lDir "${lib}" DIRECTORY)
 		if (lDir)
+			# TODO: replace with file(REAL_PATH ...) in CMake >= 3.19
 			get_filename_component(lDir "${lDir}" REALPATH)
+			list(APPEND _libdirs "${lDir}")
 		endif()
-		get_filename_component(lFile "${l}" NAME_WE)
+		# TODO: replace with cmake_path(...) in CMake >= 3.19
+		get_filename_component(lFile "${lib}" NAME_WE)
 		string(REPLACE "${_SHARED_LIBRARY_PREFIX}" "" lFile "${lFile}")
-		list(APPEND libdirs "${lDir}")
-		list(APPEND libs "${lFile}")
+		list(APPEND _libs_result "${lFile}")
 	endforeach()
-	list(REMOVE_DUPLICATES libdirs)
-	list(REMOVE_AT libdirs 0) # as it is the base libdir and will be handled separately
+	list(REMOVE_DUPLICATES _libdirs)
+	list(REMOVE_AT _libdirs 0) # as it is the base libdir and will be handled separately
 
-	set(${_library_dirs_var} "${libdirs}" PARENT_SCOPE)
-	set(${_library_names_var} "${libs}" PARENT_SCOPE)
+	set(${ARG_RESULT_LIB_DIRS} "${_libdirs}" PARENT_SCOPE)
+	set(${ARG_RESULT_LIB_NAMES} "${_libs_result}" PARENT_SCOPE)
 endfunction()
 
-split_library_dirs("${_TARGET_INTERFACE_LINK_LIBRARIES}" "${CMAKE_INSTALL_PREFIX}/${_INSTALL_LIBDIR}" _lib_dirs _library_names)
-cmake_list_to_pkg_config(_libs "${_library_names}" "-l")
+# create string value to be appended to the `Libs:` field of the pkg-config file
+split_library_dirs(
+	RESULT_LIB_DIRS "_lib_dirs"
+	RESULT_LIB_NAMES "_library_names"
+	LIBS "${_TARGET_INTERFACE_LINK_LIBRARIES}"
+	BASE_LIB_DIR "${CMAKE_INSTALL_PREFIX}/${_INSTALL_LIBDIR}"
+)
+
+cmake_list_to_pkg_config_value_str(
+	RESULT_VAR _libs
+	LIST "${_library_names}"
+	PREFIX "-l"
+)
+# TODO: Ugly OpenSSL hack.
+# For some reason the target names do not get resolved into lib names.
```

Comment:
They get resolved in the current master, so you probably broke it somewhere. Maybe double check that you enquoted strings correctly?

Reply:
resolved